packages/icon/src/vaadin-icon.js

```diff
@@ -254,6 +273,49 @@ class Icon extends ThemableMixin(
     }
   }
 
+  async __srcChanged(src) {
+    if (!src) {
+      this.svg = null;
+      return;
+    }
+
+    // Need to add the "icon" attribute to avoid issues as described in
+    // https://github.com/vaadin/web-components/issues/6301
+    this.icon = '';
+
+    if (src.includes('#')) {
+      this.svg = svg`<use href="${src}"/>`;
+    } else {
+      try {
+        const svgData = await this.__fetchSVG(src);
+        if (!Icon.__domParser) {
+          Icon.__domParser = new DOMParser();
+        }
+        const parsedResponse = Icon.__domParser.parseFromString(svgData, 'text/html');
+
+        const svgElement = parsedResponse.querySelector('svg');
+        if (!svgElement) {
+          throw new Error(`SVG element not found on path: ${src}`);
+        }
+
+        this.__viewBox = svgElement.getAttribute('viewBox');
+        this.svg = unsafeSvgLiteral(svgElement.innerHTML);
+      } catch (e) {
+        console.error(e);
+        this.svg = null;
+      }
+    }
+  }
+
+  async __fetchSVG(path) {
```

Comment:
Missing `/** @private */`. However, now that you're no longer stubbing this function in the tests, it would be an option to inline this logic to `__srcChanged`.